File: kattis/easy/java/bela.java
package easy;

import java.util.Scanner;

public class bela {

	private static int getValue(String card, Boolean dominant) {
		return switch (card) {
			case "A" -> 11;
			case "K" -> 4;
			case "Q" -> 3;
			case "J" -> dominant ? 20 : 2;
			case "T" -> 10;
			case "9" -> dominant ? 14 : 0;
			default -> 0;
		};
		// note: kattis kan ikke kjøre return switch, sendt inn er gammel switch
	}

	public static void main(String[] args) {
		try (Scanner scanner = new Scanner(System.in)) {
			int n = scanner.nextInt();
			String main_suit = scanner.next();
			scanner.nextLine();

			int sum = 0;
			for (int i=0; i < 4*n; i++) {
				String line = scanner.nextLine();
				String card = String.valueOf(line.charAt(0));
				String suit = String.valueOf(line.charAt(1));

				if (suit.equals(main_suit))
					sum += getValue(card, true);
				else sum += getValue(card, false);
			}
			System.out.println(sum);
		}
	}
}

File: kattis/easy/ccpp/bela.cpp
//
// Created by BBJ on 03.01.2024.
//
#include <iostream>

int getValue(const char& card, bool dominant) {
    if (card == 'A') return 11;
    if (card == 'K') return 4;
    if (card == 'Q') return 3;
    if (card == 'J') {if (dominant) return 20; else return 2;}
    if (card == 'T') return 10;
    if (card == '9') {if (dominant) return 14; else return 0;}
    return 0;
}

int main() {
    int n;
    char main_suit;
    std::cin >> n >> main_suit;

    int sum = 0;
    for (auto i=0; i < 4*n; i++) {
        std::string line; std::cin >> line;
        char card = line[0];
        char suit = line[1];

        if (suit == main_suit) sum += getValue(card, true);
        else sum += getValue(card, false);
    }
    std::cout << sum;
    return 0;
}